src/dialogs/backup/dialog-backup-upload.ts

```diff
@@ -0,0 +1,134 @@
+import { mdiClose, mdiFolderUpload } from "@mdi/js";
+import { css, CSSResultGroup, html, LitElement, nothing } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import { fireEvent } from "../../common/dom/fire_event";
+import "../../components/ha-file-upload";
+import "../../components/ha-header-bar";
+import "../../components/ha-icon-button";
+import { haStyleDialog } from "../../resources/styles";
+import type { HomeAssistant } from "../../types";
+import { BackupUploadDialogParams } from "./show-dialog-backup-upload";
+import { HassDialog } from "../make-dialog-manager";
+import { showAlertDialog } from "../generic/show-dialog-box";
+import { uploadBackup } from "../../data/backup";
+
+const SUPPORTED_FORMAT = "application/x-tar";
+
+@customElement("dialog-backup-upload")
+export class DialogBackupUpload
+  extends LitElement
+  implements HassDialog<BackupUploadDialogParams>
+{
+  @property({ attribute: false }) public hass?: HomeAssistant;
+
+  @state() private _dialogParams?: BackupUploadDialogParams;
+
+  @state() private _uploading = false;
+
+  public async showDialog(
+    dialogParams: BackupUploadDialogParams
+  ): Promise<void> {
+    this._dialogParams = dialogParams;
+    await this.updateComplete;
+  }
+
+  public closeDialog(): void {
+    this._dialogParams = undefined;
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+  }
+
+  protected render() {
+    if (!this._dialogParams || !this.hass) {
+      return nothing;
+    }
+
+    return html`
+      <ha-dialog
+        open
+        scrimClickAction
+        escapeKeyAction
+        hideActions
+        heading="Upload backup"
+        @closed=${this.closeDialog}
+      >
+        <div slot="heading">
+          <ha-header-bar>
+            <span slot="title"> Upload backup </span>
+            <ha-icon-button
+              .label=${this.hass.localize("ui.common.close")}
+              .path=${mdiClose}
+              slot="actionItems"
+              dialogAction="cancel"
+              dialogInitialFocus
+            ></ha-icon-button>
+          </ha-header-bar>
+        </div>
+        <ha-file-upload
+          .hass=${this.hass}
+          .uploading=${this._uploading}
+          .icon=${mdiFolderUpload}
+          accept=${SUPPORTED_FORMAT}
+          label="Upload a backup"
+          supports="Supports .tar files"
+          @file-picked=${this._uploadFile}
+        ></ha-file-upload>
+      </ha-dialog>
+    `;
+  }
+
+  private async _uploadFile(ev: CustomEvent<{ files: File[] }>): Promise<void> {
+    const file = ev.detail.files[0];
+
+    if (file.type !== SUPPORTED_FORMAT) {
+      showAlertDialog(this, {
+        title: "Unsupported file format",
+        text: "Please choose a Home Assistant backup file (.tar)",
+        confirmText: "ok",
+      });
+      return;
+    }
+    this._uploading = true;
+    try {
+      await uploadBackup(this.hass!, file);
+      fireEvent(this, "backup-file-uploaded");
```

Comment:
```suggestion
       this._params.backupFileUploaded()
```

We should use a callback instead of a event.
The dialog is added at the root level of the DOM so I would be impossible for the component that open the dialog to catch it.